File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/engine/TouchMessage.java
/**
 * 
 */
package com.abubusoft.xenon.engine;

import com.abubusoft.xenon.core.collections.PooledObject;

/**
 * @author Francesco Benincasa
 *
 */
public class TouchMessage implements PooledObject {

	public TouchType type;
	
	public float x;
	
	public float y;

	@Override
	public void initializePoolObject() {
		
		
	}

	@Override
	public void finalizePoolObject() {
		type=null;
		
		x=0f;
		y=0f;
		
	}
	
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/engine/TouchMessage.kt
/**
 *
 */
package com.abubusoft.xenon.engine

import com.abubusoft.xenon.core.collections.PooledObject

/**
 * @author Francesco Benincasa
 */
class TouchMessage : PooledObject {
    var type: TouchType? = null
    var x = 0f
    var y = 0f
    override fun initializePoolObject() {}
    override fun finalizePoolObject() {
        type = null
        x = 0f
        y = 0f
    }
}